dora/core/common/src/main/java/alluxio/uri/UfsUrl.java

```diff
@@ -1,18 +1,87 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
 package alluxio.uri;
 
 import alluxio.AlluxioURI;
 import alluxio.grpc.UfsUrlMessage;
+
+import com.google.common.base.Preconditions;
 import org.apache.logging.log4j.util.Strings;
 
+import java.util.Arrays;
 import java.util.List;
+import java.util.Objects;
 
 public class UfsUrl {
+
+  // TODO(Tony Sun): replace each xx_separator by the static variables.
+  public static final String SCHEME_SEPERATOR = "://";
+  public static final String PATH_SEPERATOR = "/";
   UfsUrlMessage mProto;
 
   public UfsUrl(UfsUrlMessage proto) {
+    Preconditions.checkArgument(proto.getPathComponentsList().size() != 0,
+        "the proto.path is empty, please check the proto first");
     mProto = proto;
   }
 
+  public UfsUrl(String ufsPath) {
+    // TODO(Tony Sun): Considering the case below:
+    //  when scheme does not exist, how to determine the scheme, or a empty scheme.
```

Comment:
I think we have two options:
1. Assume the input path is a relative path to the Alluxio root
2. Assume the input path is local file and prefix that with "file://"

I prefer option 1